screen-orientation/orientation-reading.html

```diff
@@ -20,6 +22,41 @@
   assert_true(angle == 0 || angle == 90 || angle == 180 || angle == 270);
 }, "Test screen.orientation default values.");
 
+promise_test(async t => {
+  await test_driver.bless("request full screen", () => {
+    return document.documentElement.requestFullscreen();
+  });
+  let primaryOrientation;
```

Comment:
Could maybe instead do:

```JS
  await screen.orientation.lock("portrait-primary");
  const primaryOrientation = screen.orientation.angle === 0 
    ? "portrait-primary"
    : "landscape-primary";
```

But I'm a bit confused... if you do:

```
await screen.orientation.lock("portrait-primary");
```

Won't that assure the orientation is "portrait-primary"? 

And even if not assured, it's probably better to the value straight from "the horse's mouth"... i.e., `primaryOrientation` should be `screen.orientation.type`, right? Or am I missing something?

Reply:
I wrote this to try to find out what `primaryOrientation` is - what we call in the screen orientation values table `primary orientation 1 `- and using this work out all the other angles as stated in the table. https://w3c.github.io/screen-orientation/#reading-the-screen-orientation
If when the screen is locked to `portrait-primary` the angle is 0 - then we know that this is `primary orientation 1`, so `secondary orientation 1` would then be `portrait-secondary` at 180 degrees.  `Primary orientation 2` would be `landscape primary `at either 90 or 270 and `secondary orientation 2` would be the remaining angle allocated to `landscape-secondary.`
If the screen is locked to `portrait-primary` and the angle is not 0, (it should either be 90 or 270) then `landscape-primary` would be `primary orientation 1 `at 0.  `Landscape-secondary` would be `secondary orientation 1` at 180.  `Portrait-primary` at 90 or 270 would be `primary-orientation 2` and so on.
This enables a more thorough check of the relationship of orientation to angle.

Reply:
Ok, yes, I get the logic and rationale, which is perfect. 

But I'm still a bit confused (heh, I should read the spec!) about the following... if we do:

```JS
await screen.orientation.lock("portrait-primary");
```

Won't the `screen.orientation.type` be "portrait-primary"? 

If yes, then it follows that `primaryOrientation` will just be  "portrait-primary".

Reply:
Yes, but we're testing for the angle not the type to decide primaryOrientation.  So once we are locked to portrait-primary - if the angle is 0 - then the primary orientation is "portrait-primary".

If the angle is 90 or 270, we know that the primary orientation is "landscape-primary" and portrait-primary is a rotation from that.

Am I confused though??